### PR TITLE
Story/quad 86

### PR DESCRIPTION
**Pull request by prasanthpriyanesan**

### CODE REVIEW

File: Quadriga/.classpath [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+			<attribute name="org.eclipse.jst.component.dependency" value="/WEB-INF/lib"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7">
+		<attributes>
+			<attribute name="maven.pomderived" value="true"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="output" path="target/classes"/>
+</classpath>
```
Comment (jdamerow) [inline Quadriga/.classpath:30]: I think we talked about that. Make sure to get the latest version of this file from develop. If you are working on a Windows computer please adjust the settings (git autocrlf).

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        return model;
+    }
+
+    /**
+     * This method is used update the Text blocks in the Public settings page
+     *
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.GET, value = "auth/workbench/{projectid}/addpublicpage1")
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:57]: bad naming: addpublicpage1

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    public @ResponseBody ResponseEntity<String> getExistingConcepts(
+            @RequestParam("data1") String data1)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:59]: what is data1?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            RoleNames.ROLE_COLLABORATOR_ADMIN,
+            RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/{projectid}/addpublicpage", method = RequestMethod.GET)
+    public ModelAndView publicPageSettingsForm(
+            @PathVariable("projectid") String projectid)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        ModelAndView model = new ModelAndView("auth/workbench/addpublicpage");
+        model.getModelMap().put("publicpage",
+                publicPageFactory.createPublicPageObject());
+        model.getModelMap().put("ppprojectid", projectid);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:47]: why `ppprojectid`?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    /**
+     * This method is used update the Text blocks in the Public settings page
+     *
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.GET, value = "auth/workbench/{projectid}/addpublicpage1")
+    public @ResponseBody ResponseEntity<String> getExistingConcepts(
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:58]: What is this method doing?

File: Quadriga/src/main/resources/locale/messages_en_US.properties [hunk 6]
```diff
@@ -142,3 +142,7 @@ account_approval.error=Sorry, the account couldn't be approved. There seems to b
 account_approval.account_roles.required=Please specify at least one role for the user.
 account_approval.action.required=Please select "Approve" or "Deny".
 
+#Public Page Settings Validation Error Messages
+title.required = Title is Mandatory
```
Comment (jdamerow) [inline Quadriga/src/main/resources/locale/messages_en_US.properties:146]: should be "Please provide a title."

File: Quadriga/src/main/resources/locale/messages_en_US.properties [hunk 7]
```diff
@@ -142,3 +142,7 @@ account_approval.error=Sorry, the account couldn't be approved. There seems to b
 account_approval.account_roles.required=Please specify at least one role for the user.
 account_approval.action.required=Please select "Approve" or "Deny".
 
+#Public Page Settings Validation Error Messages
+title.required = Title is Mandatory
+description.required = Description is Mandatory
```
Comment (jdamerow) [inline Quadriga/src/main/resources/locale/messages_en_US.properties:147]: should be "Please provide a description."

File: Quadriga/src/main/resources/locale/messages_en_US.properties [hunk 8]
```diff
@@ -142,3 +142,7 @@ account_approval.error=Sorry, the account couldn't be approved. There seems to b
 account_approval.account_roles.required=Please specify at least one role for the user.
 account_approval.action.required=Please select "Approve" or "Deny".
 
+#Public Page Settings Validation Error Messages
+title.required = Title is Mandatory
+description.required = Description is Mandatory
+order.required = Order is Mandatory
```
Comment (jdamerow) [inline Quadriga/src/main/resources/locale/messages_en_US.properties:148]: should be "On what position should the tex be? Possible values are 1, 2, or 3."

File: Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+								$("#title_err1").html(''); 
+							}
+							if(desc.length < 1){
+								$("#desc_err1").html("Description Is Mandatory");
+								mandatory =1;
+							}else{
+								$("#desc_err1").html('');
+							}
+							if(order == 0){
+								$("#order_err1").html("Order Is Mandatory");
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp:30]: Should be the same messages as above.

File: Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+										type : "GET",
+										url : "${pageContext.servletContext.contextPath}/auth/workbench/${ppprojectid}/addpublicpage1",
+										data : {
+											data1 : JSON.stringify(data)
+										},
+										success : function(e) {
+											$("#title1").val('');
+											$("#description1").val('');
+											$("#order1").val('');
+										},
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp:52]: Why are you setting the values to an empty string here?

File: Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+							if(mandatory){
+								return false;
+							}
+							data["title"] = $("#title1").val();
+							data["desc"] = $("#description1").val();
+							data["order"] = $("#order1").val();
+							$
+									.ajax({
+										type : "GET",
+										url : "${pageContext.servletContext.contextPath}/auth/workbench/${ppprojectid}/addpublicpage1",
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp:44]: What is this doing? If it is sending/saving data, it should be a POST request.

File: Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+											$("#title3").val('');
+											$("#description3").val('');
+											$("#order3").val('');
+										},
+										error : function(e) {
+											console.log("ERROR: ", e);
+										}
+									});
+ 						});
+	});
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp:162]: Do you have the same function three times here? No code duplications! Have it once, but bind it several times.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java (hunk 3 above)
Comment (prasanthpriyanesan) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:59]: this data will contain the the updated information about the text blocks which will be stored in database(QUAD-111)

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java (hunk 4 above)
Comment (prasanthpriyanesan) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:47]: will change the name to publicpageprojectid

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java (hunk 5 above)
Comment (prasanthpriyanesan) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/PublicPageController.java:58]: this is a dummy method created for this story, which will be used by thara to store the data coming from public page settings page into databsae

File: Quadriga/src/main/resources/locale/messages_en_US.properties (hunk 6 above)
Comment (prasanthpriyanesan) [inline Quadriga/src/main/resources/locale/messages_en_US.properties:146]: will change them accordingly

File: Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp (hunk 12 above)
Comment (prasanthpriyanesan) [inline Quadriga/src/main/webapp/WEB-INF/views/workbench/addpublicpage.jsp:162]: each function is for three separate save buttons. I'll check about this with you

**Pull request closed without merging**